### PR TITLE
fix: duplicates in field names

### PR DESCRIPTION
**Pull request by harisarang**
fix: #328

### CODE REVIEW

**Comment by harisarang** [comment]
fix: #328

File: include/field.h [hunk 1]
```diff
@@ -259,6 +263,12 @@ struct field {
                                             "` but is not found in the schema.");
         }
 
+        LOG(WARNING) << "Fields size: " << fields.size() << " Field set size: " << field_names.size();
+        if(field_names.size() != fields.size()) {
+            return Option<bool>(400, "Duplicates are not allowed in `fields`");
```
Comment (kishorenc) [inline include/field.h:268]: Let's change the error message to: `There are duplicate field names in the schema.`

File: test/collection_specific_test.cpp [hunk 2]
```diff
@@ -1991,3 +1991,11 @@ TEST_F(CollectionSpecificTest, SingleHyphenInQueryNotToBeTreatedAsExclusion) {
     ASSERT_EQ(1, results["hits"].size());
     collectionManager.drop_collection("coll1");
 }
+
+TEST_F(CollectionSpecificTest, DuplicateFieldsNotAllowed) {
+    std::vector<field> fields = {field("title", field_types::STRING, false),
+                                 field("title", field_types::INT32, true),};
+    Option<Collection*> response = collectionManager.create_collection("collection", 1, fields);
+
+    ASSERT_FALSE(response.ok());
```
Comment (kishorenc) [inline test/collection_specific_test.cpp:2000]: Need to assert on the error message too by checking on `response.error()`

File: include/field.h [hunk 3]
```diff
@@ -259,6 +263,12 @@ struct field {
                                             "` but is not found in the schema.");
         }
 
+        LOG(WARNING) << "Fields size: " << fields.size() << " Field set size: " << field_names.size();
```
Comment (kishorenc) [inline include/field.h:266]: Debugging log? To be removed.

**Review by kishorenc — CHANGES_REQUESTED** [review]

**Review by kishorenc — APPROVED** [review]

**Pull request merged**